api_tests/src/cnd/swaps_payload.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * These are the payload for the `/swaps/` REST API that supports halight swaps (lightning) and that will replace
+ * the `/swaps/rfc003` endpoint once all other swaps pair are supported.
+ */
+
+import { Asset, Ledger, Peer } from "./cnd";
+import { Action, EmbeddedRepresentationSubEntity, Entity, Link } from "./siren";
+
+/**
+ * The parameters that the two parties have agreed on for swap execution.
+ *
+ * The swap request is sent by the party in the role of Alice to initiate a swap with Bob.
+ */
+export interface SwapRequest {
+    alpha_ledger: Ledger;
+    alpha_asset: Asset;
+    beta_ledger: Ledger;
+    beta_asset: Asset;
+    alpha_expiry?: number;
+    beta_expiry?: number;
+    alpha_ledger_refund_identity?: string;
+    beta_ledger_redeem_identity?: string;
+    peer: Peer;
+}
+
+interface CoreRequestBody<A, B> {
+    alpha: A;
+    beta: B;
+    role: "Alice" | "Bob";
+    peer: Peer;
+}
+
+interface RequestParams {
+    amount: string;
+    identity: string;
+}
+
+interface Herc20Request {
+    contract_address: string;
+    absolute_expiry: number;
+}
+
+interface HalightRequest {
```

Comment:
Isn't this already defined in the e2e tests? I thought we dupe it already as we were doing iterations on the halight endpoints.